src/config.ts

```diff
@@ -90,37 +90,38 @@ const config = {
       certDuration: 31536000,
     },
     networkConfig: {
-      // ForwardProxy
-      forwardHost: '127.0.0.1' as Host,
-      forwardPort: 0 as Port,
-      proxyHost: '0.0.0.0' as Host,
-      proxyPort: 0 as Port,
       // GRPCServer for agent service
       agentHost: '127.0.0.1' as Host,
       agentPort: 0 as Port,
+      ipv6Only: false,
       // RPCServer for client service
       clientHost: '127.0.0.1' as Host,
       clientPort: 0 as Port,
-      // Times and limits for client communication
+      // Websocket server config
+      maxReadableStreamBytes: 1_000_000_000, // About 1 GB
       connectionIdleTimeoutTime: 120, // 2 minutes
       pingIntervalTime: 1_000, // 1 second
       pingTimeoutTime: 10_000, // 10 seconds
+      // RPC config
+      clientParserBufferByteLimit: 1_000_000, // About 1MB
       handlerTimeoutTime: 60_000, // 1 minute
       handlerTimeoutGraceTime: 2_000, // 2 seconds
-      maxReadableStreamBytes: 1_000_000_000, // About 1 GB
-      clientParserBufferByteLimit: 1_000_000, // About 1MB
     },
-    proxyConfig: {
-      connConnectTime: 2000,
-      connKeepAliveTimeoutTime: 20000,
-      connEndTime: 1000,
-      connPunchIntervalTime: 100,
-      connKeepAliveIntervalTime: 1000,
+    quicServerConfig: {
+      keepAliveIntervalTime: 10_000, // 10 seconds
+      maxIdleTimeout: 60_000, // 1 minute
+    },
+    quicClientConfig: {
+      keepAliveIntervalTime: 10_000, // 10 seconds
+      maxIdleTimeout: 60_000, // 1 minute
     },
     nodeConnectionManagerConfig: {
       connConnectTime: 2000,
       connTimeoutTime: 60000,
       initialClosestNodes: 3,
+      pingTimeout: 2000,
+      holePunchTimeout: 4000,
+      holePunchInitialInterval: 250,
```

Comment:
It works fine? `PingTimeout` sets the connection creation time when testing a connection. hole punching works on the `QUICSocket` directly for sending the packets directly and doesn't touch any `QUICConnections`. 

Only the `pingTimeout` is related to the max idle timeout, and it only needs to be less than the idle timeout.

Reply:
I believe by convention we should be naming all time variables with the suffix "Time". So `pingTimeoutTime` or `holePunchTimeoutTime` and `holePunchInitialIntervalTime`. This is to differentiate it from a specific "object" that manages the timeout or certain functions that create a timeout... etc etc.

> Only the `pingTimeout` is related to the max idle timeout, and it only needs to be less than the idle timeout.

Is the condition checked here?

And also all this should be documented as inline commentary here.

Actually I'm kind of confused, what does `pingTimeout` even mean here at this point? Is this supposed to be part of our regular keep alive?

I mean for agent to agent communication my understanding is that:

1. There's a max idle timeout necessary to indicate when we consider the connection to be dead.
2. There should be a keep alive interval time that sends ping/pongs. But in the case of quic, this is managed "intelligently" within quic, this should still be a configuration variable.
3. Hole punch timeout and hole punch interval is independent of connection timeouts.
4. Then what is `pingTimeout`...?

All of these time variables should be prefixed by the thing it's used for. So we can say:

* `connectionIdleTimeoutTime`
* `connectionKeepAliveIntervalTime`
* `connectionHolePunchIntervalTime`
* `connectionHolePunchTimeoutTime`.

We may need to distinguish any timeouts that are between a "node connection" and the "transport connection" and "stream" where they would all potentially have different timeout times or interval times.

Reply:
`pingTimeout` is used for the connection timeout when checking a connection can be made. so generally used when we ping a node to check it can be reached.

I'll update the names.

Reply:
I updated the hole punch names, but the `keepAliveIntervalTime` and `maxIdleTimeout` are names set by js-quic and we're just passing that along.

Reply:
Usually even if we are passing names through we still either namespace them or prefix them by the terminal domain.